include/dgl/runtime/ndarray.h

```diff
@@ -127,6 +127,11 @@ class NDArray {
   inline const DLTensor* operator->() const;
   /*! \return True if the ndarray is contiguous. */
   bool IsContiguous() const;
+  /*! \return the data pointer with type. */
+  template <typename T>
+  inline T* Ptr() const {
+    return static_cast<T*>(operator->()->data);
```

Comment:
I suggest returning a nullptr when the ndarray itself is a NullArray, see #1644 .